src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -431,44 +431,41 @@ trait SubmissionComponent {
 
       def deleteAction(workspaceId: UUID) = {
 
-        def workspaceSubmissionQuery(workflow: SQLActionBuilder, join: SQLActionBuilder) = {
-          concatSqlActions(sql" inner join ", workflow, sql" w on ", join,
-            sql" inner join SUBMISSION s on w.submission_id=s.id where s.workspace_id=${workspaceId}")
+        def deleteSubmissionAttributes(workflowTable: String, columnId: String) = {
+          sqlu""" delete t from SUBMISSION_ATTRIBUTE t
+                 inner join SUBMISSION_VALIDATION sv on sv.id=t.owner_id
+                 inner join #$workflowTable w on w.id=sv.#$columnId
+                 inner join SUBMISSION s on s.id=w.submission_id
+                 where s.workspace_id=$workspaceId
+            """
         }
 
-        val tables: Seq[SQLActionBuilder] = Seq(sql"WORKFLOW", sql"WORKFLOW_FAILURE")
+        def deleteTwoJoins(tableFrom: String, workflowTable: String, columnId: String) = {
+          sqlu"""delete t from #$tableFrom t
+                 inner join #$workflowTable w on w.id=t.#$columnId
+                 inner join SUBMISSION s on w.submission_id=s.id
+                 where s.workspace_id=$workspaceId
+              """
+        }
+
+        val tables: Seq[String] = Seq(s"WORKFLOW", s"WORKFLOW_FAILURE")
 
         DBIO.seq(
-          // delete submission attributes, workflow messages, and submission validations for workflows and
-          // workflow failures
-          concatSqlActions(
-            sql"""delete sa from SUBMISSION_ATTRIBUTE sa
-                  inner join SUBMISSION_VALIDATION sv on sv.id=sa.owner_id
-              """, workspaceSubmissionQuery(sql"WORKFLOW", sql"sv.workflow_id=w.id")).as[Int],
-          concatSqlActions(
-            sql"""delete sa from SUBMISSION_ATTRIBUTE sa
-                  inner join SUBMISSION_VALIDATION sv on sv.id=sa.owner_id
-              """, workspaceSubmissionQuery(sql"WORKFLOW_FAILURE", sql"sv.workflow_failure_id=w.id")).as[Int],
-          concatSqlActions(sql"delete wm from WORKFLOW_MESSAGE wm ",
-            workspaceSubmissionQuery(sql"WORKFLOW", sql"wm.workflow_id=w.id")).as[Int],
-          concatSqlActions(sql"delete wm from WORKFLOW_MESSAGE wm ",
-            workspaceSubmissionQuery(sql"WORKFLOW_FAILURE", sql"wm.workflow_id=w.id")).as[Int],
-          concatSqlActions(sql"delete sv from SUBMISSION_VALIDATION sv",
-            workspaceSubmissionQuery(sql"WORKFLOW", sql"sv.workflow_id=w.id")).as[Int],
-          concatSqlActions(sql"delete sv from SUBMISSION_VALIDATION sv",
-            workspaceSubmissionQuery(sql"WORKFLOW_FAILURE", sql"sv.workflow_failure_id=w.id")).as[Int],
-          concatSqlActions(sql"delete we from WORKFLOW_ERROR we",
-            workspaceSubmissionQuery(sql"WORKFLOW_FAILURE", sql"w.id=we.workflow_failure_id")).as[Int]
+          deleteSubmissionAttributes(s"WORKFLOW", s"workflow_id"),
```

Comment:
Very nice.

Can also remove the `s` here.

I'd prefer a name like `deleteFromTable` instead of `deleteTwoJoins`.